### ISSUE TITLE
vagrant --help displays a rubygems error

### ISSUE THREAD

**Issue by mfdj** [mitchellh/vagrant#8519]
### Vagrant version

```sh
$ vagrant --version
1.9.4
```

### Host operating system

```sh
$ sw_ver
ProductName:     Mac OS X
ProductVersion:   10.12.4
BuildVersion:    16E195
```

### Expected behavior

`vagrant` and `vagrant --help` prints a list of commands

### Actual behavior

A rubygems error occurs.

<pre>
/opt/vagrant/embedded/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- vagrant-share/helper/api (LoadError)
from /opt/vagrant/embedded/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
from /opt/vagrant/embedded/gems/gems/vagrant-share-1.1.7/lib/vagrant-share/activate.rb:244:in `<encoded>'
from /opt/vagrant/embedded/gems/gems/vagrant-share-1.1.7/lib/vagrant-share/activate.rb:16:in `RGLoader_load'
from /opt/vagrant/embedded/gems/gems/vagrant-share-1.1.7/lib/vagrant-share/activate.rb:16:in `<top (required)>'
from /opt/vagrant/embedded/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
from /opt/vagrant/embedded/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
from /opt/vagrant/embedded/gems/gems/vagrant-share-1.1.7/lib/vagrant-share.rb:23:in `block in <class:Plugin>'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/cli.rb:75:in `call'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/cli.rb:75:in `block (2 levels) in help'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/registry.rb:49:in `block in each'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/registry.rb:48:in `each'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/registry.rb:48:in `each'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/cli.rb:69:in `block in help'
from /opt/vagrant/embedded/lib/ruby/2.2.0/optparse.rb:917:in `initialize'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/cli.rb:57:in `new'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/cli.rb:57:in `help'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/cli.rb:32:in `execute'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/environment.rb:308:in `cli'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/bin/vagrant:127:in `<main>'
</pre>

### Steps to reproduce

1. Install vagrant from the web
2. Run `vagrant --help`

**Comment by piotr-cz** [comment]
This is a 1.9.4 thing,
it happens when running just `vagrant` too

**Comment by teresaejunior** [comment]
I think the latest Vagrant build is just incomplete. Same error on Vagrant 1.9.4, Ubuntu 17.04.

**Comment by mattyhansen** [comment]
Also happening on Windows 10 (vagrant 1.9.1 on other machine is working okay)

**Comment by adiachenko** [comment]
I concur that Vagrant 1.9.4 doesn't work on Windows 10.

**Comment by xjchengo** [comment]
I upgraded Vagrant to 1.9.4 on Windows 8. Same error occurs.
```
C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file
-- vagrant-share/helper/api (LoadError)
        from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-share-1.1.7/lib/vagrant-share/activate.rb:244:in `<encoded>
'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-share-1.1.7/lib/vagrant-share/activate.rb:16:in `RGLoader_l
oad'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-share-1.1.7/lib/vagrant-share/activate.rb:16:in `<top (requ
ired)>'
        from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-share-1.1.7/lib/vagrant-share.rb:23:in `block in <class:Plu
gin>'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/cli.rb:75:in `call'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/cli.rb:75:in `block (2 levels) in help'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/registry.rb:49:in `block in each'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/registry.rb:48:in `each'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/registry.rb:48:in `each'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/cli.rb:69:in `block in help'
        from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/optparse.rb:917:in `initialize'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/cli.rb:57:in `new'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/cli.rb:57:in `help'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/cli.rb:32:in `execute'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/environment.rb:308:in `cli'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.9.4/bin/vagrant:127:in `<main>'
```

**Comment by chrisroberts** [comment]
To fix this error:

```
$ vagrant plugin install vagrant-share --plugin-version 1.1.8
```

**Comment by mfdj** [comment]
Can confirm that upgrading `vagrant-share` from 1.1.7 to 1.1.8 stops the rubygems error.

**Comment by DnZmfr** [comment]
I also confirm @chrisroberts' solution 👍  Thanks :)

**Comment by rockts** [comment]
非常感谢，这个命令解决了问问难题

**Comment by amontalban** [comment]
Any idea if a 1.9.5 version is going to be released quickly to address this?

**Comment by adiachenko** [comment]
I second the question, 1.9.4 is still posted on official site, this is just bad considering there in no official notice about this bug and you need to go out of your way to download the older versions.

**Comment by stephenconnolly1** [comment]
I'm airgapped so can't update offline to get the older version :(  How do1.1.8 plugin and install?

**Comment by chrisroberts** [comment]
@stephenconnolly1 First fetch the gem:

```
$ gem fetch --clear-sources --source https://gems.hashicorp.com vagrant-share
```

move gem to offline machine, then install as plugin:

```
$ vagrant plugin install ./vagrant-share-1.1.8.gem
```

**Comment by belcebus** [comment]
vagrant plugin install vagrant-share --plugin-version 1.1.8 only fix vagrant --help and vagrant command, but when starting virtual machine from Windows 7 Host the error still happens:

    development: Warning: Remote connection disconnect. Retrying...
==> development: Forcing shutdown of VM...
==> development: Destroying VM and associated drives...
C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/transport/server_version.rb:54:in `readpartial': An established connection was aborted by the software in your host machine. (Errno::ECONNABORTED)
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/transport/server_version.rb:54:in `block (2 levels) in negotiate!'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/transport/server_version.rb:52:in `loop'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/transport/server_version.rb:52:in `block in negotiate!'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/transport/server_version.rb:50:in `loop'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/transport/server_version.rb:50:in `negotiate!'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/transport/server_version.rb:32:in `initialize'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/transport/session.rb:84:in `new'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/transport/session.rb:84:in `initialize'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh.rb:233:in `new'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh.rb:233:in `start'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.9.4/plugins/communicators/ssh/communicator.rb:397:in `block (2 levels) in connect'
        from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/timeout.rb:88:in `block in timeout'
        from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/timeout.rb:32:in `block in catch'
        from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/timeout.rb:32:in `catch'
        from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/timeout.rb:32:in `catch'
        from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/timeout.rb:103:in `timeout'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.9.4/plugins/communicators/ssh/communicator.rb:371:in `block in connect'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/util/retryable.rb:17:in `retryable'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.9.4/plugins/communicators/ssh/communicator.rb:370:in `connect'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.9.4/plugins/communicators/ssh/communicator.rb:68:in `block in wait_for_ready'
        from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/timeout.rb:88:in `block in timeout'
        from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/timeout.rb:32:in `block in catch'
        from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/timeout.rb:32:in `catch'
        from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/timeout.rb:32:in `catch'
        from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/timeout.rb:103:in `timeout'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.9.4/plugins/communicators/ssh/communicator.rb:46:in `wait_for_ready'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/action/builtin/wait_for_communicator.rb:16:in `block in call'

**Comment by chrisroberts** [comment]
@belcebus See #8520.

**Comment by belcebus** [comment]
Thanks! That works for me!

Regards

**Comment by UmanShahzad** [comment]
Christrobert's solution works. Was wanting to try this solution for the first time today and ran into the error. Would be very helpful if the official version was updated lest confusion ensue..

**Comment by danstreeter** [comment]
I can confirm that the advice by @chrisroberts above to perform:
`vagrant plugin install vagrant-share --plugin-version 1.1.8`
has worked for me on a recently upgraded to Vagrant 1.9.4 Win10 machine.

Thanks.

**Comment by cgatesman** [comment]
The workaround works, but as soon as you do a `vagrant plugin update` it reverts back to 1.1.7.

**Comment by SPCTR13** [comment]
Glad I found this thread. I had a problem with "Vagrant up" on 1.9.3... I am new, so I reinstalled with Vagrant 1.9.4, imagine my surprise when I got that error.  Good that Vagrant plugin update worked on Win10/64.  Thanks.

**Comment by WestleyArgentum** [comment]
Bump – it would be really nice if this went out asap

**Comment by uoryon** [comment]
thx

**Comment by techmaniack** [comment]
vagrant: 1.9.4 
osx: 10.12.4

```
➜  ~ vagrant
/opt/vagrant/embedded/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- vagrant-share/helper/api (LoadError)
	from /opt/vagrant/embedded/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /opt/vagrant/embedded/gems/gems/vagrant-share-1.1.7/lib/vagrant-share/activate.rb:244:in `<encoded>'
	from /opt/vagrant/embedded/gems/gems/vagrant-share-1.1.7/lib/vagrant-share/activate.rb:16:in `RGLoader_load'
	from /opt/vagrant/embedded/gems/gems/vagrant-share-1.1.7/lib/vagrant-share/activate.rb:16:in `<top (required)>'
	from /opt/vagrant/embedded/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
```

**Comment by ochetski** [comment]
Confirmed that fresh instal was fixed by:
``vagrant plugin install vagrant-share --plugin-version 1.1.8``

**Comment by sergeyklay** [comment]
@chrisroberts There is reason to not release bug fix asap?

**Comment by verhulstm** [comment]
the problem also exists for vagrant v1.9.3

**Comment by bulat-nugmanov** [comment]
Like others mentioned, issue is fixed on Win 10 by updating the vagrant-share plugin version - hopefully this is fixed in next release

**Comment by eannastew** [comment]
I'm on Windows 10.
I tried vagrant plugin install vagrant-share --plugin-version 1.1.8.
It didn't work.

I then ran a vagrant plugin update
vagrant-share reverted to 1.1.7
vagrant-hostmanager upgraded to 1.8.6 (from 1.8.5)
and vagrant-vbguest upgraded to 0.14.2 (from 0.13.0)
All working now.
Hope this helps.

**Comment by olojiang** [comment]
Thanks, it works fine with: `vagrant plugin install vagrant-share --plugin-version 1.1.8`

**Comment by jay771993** [comment]
`vagrant plugin install vagrant-share --plugin-version 1.1.8` That work on osx 10.12.4

**Comment by helieting** [comment]
`vagrant plugin install vagrant-share --plugin-version 1.1.8`   work on osx 10.11.6

**Comment by xRomZak** [comment]
`vagrant plugin install vagrant-share --plugin-version 1.1.8` work on Windows 8 + Babun

**Comment by adiachenko** [comment]
Hey, guys, it's not a census, I'm pretty sure everybody have already caught on the fact that this fixes the issue. 😄

**Comment by l0010o0001l** [comment]
:P They are happy! @adiachenko

**Comment by miroswan** [comment]
Also, you don't have to go that far back to get a working version. Just a heads up.

**Comment by chrisroberts** [comment]
Fixed in 1.9.5 release. Cheers!

**Comment by piotr-cz** [comment]
Confirming it's been fixed in v1.9.5.
Too bad it's not noted in [CHANGELOG.md](https://github.com/mitchellh/vagrant/blob/master/CHANGELOG.md)

**Comment by settermjd** [comment]
Thanks @chrisroberts for [your comment](https://github.com/hashicorp/vagrant/issues/8519#issuecomment-297413831). It worked perfectly. Fwiw, my issue arose after upgrading from macOS Sierra to High Sierra.
